src/mui/components/nested-dropdown/NestedDropdown.tsx

```diff
@@ -0,0 +1,196 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import { ChevronLeft } from "@mui/icons-material";
+import ChevronRight from "@mui/icons-material/ChevronRight";
+import Box from "@mui/material/Box";
+import ClickAwayListener from "@mui/material/ClickAwayListener";
+import Divider from "@mui/material/Divider";
+import Grow from "@mui/material/Grow";
+import ListItem from "@mui/material/ListItem";
+import MenuList from "@mui/material/MenuList";
+import Paper from "@mui/material/Paper";
+import Popper, { PopperPlacementType, PopperProps } from "@mui/material/Popper";
+import { useTheme } from "@mui/material/styles";
+import Typography from "@mui/material/Typography";
+import useMediaQuery from "@mui/material/useMediaQuery";
+import React, { useCallback, useRef, useState } from "react";
+
+import { DefaultDropdownButton } from "../dropdown/DefaultDropdownButton";
+import { NestedDropdownItem } from "./NestedDropdownItem";
+
+export interface NestedDropdownAction {
+    id: string;
+    disabled: boolean;
+    shouldMenuStayOpened?: boolean;
+    key?: string;
+    isActive?: boolean;
+    isShown?: boolean;
+    isSelected?: boolean;
+    isDivider?: boolean;
+    onClick?: (action: NestedDropdownAction) => void;
+    actions?: NestedDropdownAction[];
+    leftIcon?: React.ReactElement;
+    content?: string;
+    rightIcon?: React.ReactElement;
+    header?: string;
+    contentObject?: React.ReactNode;
+}
+
+export interface NestedDropdownProps {
+    id?: string;
+    popperProps?: {
+        id: string;
+        modifiers?: PopperProps["modifiers"];
+        "data-popper-id"?: string;
+    };
+    buttonContent?: string;
+    actions: NestedDropdownAction[];
+    children?: React.ReactNode | React.ReactNode[];
+    paperPlacement?: PopperPlacementType;
+    className?: string;
+    disabled?: boolean;
+    header?: string;
+    contentObject?: React.ReactNode;
+}
+
+// TODO: discuss turning this into fully reusable component, renaming to Dropdown? and changing existing implementations to use that.
+/**
+ * @summary Nested dropdown component can be used as a default dropdown component,
+ *  with options to make it nested and display menu in a flexible way:
+ *  header - optional header text for the dropdown
+ *  [actions: leftIcon, content, rightIcon] - optional actions array which will be converted to dropdown menu items.
+ *  contentObject - optional object to display custom content layout
+ *  divider - optional divider to separate dropdown menu items
+ */
+export default function NestedDropdown({
+    id,
+    actions,
+    buttonContent,
+    popperProps = {
+        id: "popper",
+    },
+    children = null,
+    disabled = false,
+    paperPlacement = "auto-start",
+    className = "",
+    header,
+}: NestedDropdownProps) {
+    const containerRef = useRef<HTMLDivElement>(null);
+    const [opened, setOpened] = useState(false);
+    const theme = useTheme();
+    const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
```

Comment:
I think it's better to move mediaQuery to host app. In different places it could require different breakpoint triggers.